src/main/java/io/confluent/kafkarest/v2/KafkaConsumerReadTask.java

```diff
@@ -0,0 +1,232 @@
+                          ConsumerWorkerReadCallback<ClientK, ClientV> callback) {
+    this.parent = parent;
+    this.maxResponseBytes = Math.min(
+        maxBytes,
+        parent.getConfig().getLong(KafkaRestConfig.CONSUMER_REQUEST_MAX_BYTES_CONFIG));
+    this.pollTimeout = timeout <= 0 ? 5000 : timeout;
+    this.callback = callback;
+    this.finished = new CountDownLatch(1);
+
+    started = parent.getConfig().getTime().milliseconds();
+    try {
+	topicState = parent.getOrCreateTopicState(topic, pollTimeout);
+
+      // If the previous call failed, restore any outstanding data into this task.
+      KafkaConsumerReadTask previousTask = topicState.clearFailedTask();
+      if (previousTask != null) {
+        this.messages = previousTask.messages;
+        this.bytesConsumed = previousTask.bytesConsumed;
+      }
+    } catch (RestException e) {
+      finish(e);
+    }
+  }
+
+  /**
+   * Performs one iteration of reading from a consumer iterator.
+   *
+   * @return true if this read timed out, indicating the scheduler should back off
+   */
+  public boolean doPartialRead() {
+    try {
+      // Initial setup requires locking, which must be done on this thread.
+      if (iter == null) {
+        parent.startRead(topicState);
+        iter = topicState.getIterator();
+
+        messages = new Vector<ConsumerRecord<ClientK, ClientV>>();
+        waitExpiration = 0;
+      }
+
+      boolean backoff = false;
+      long roughMsgSize = 0;
+
+      long startedIteration = parent.getConfig().getTime().milliseconds();
+      final int requestTimeoutMs = parent.getConfig().getInt(
+          KafkaRestConfig.CONSUMER_REQUEST_TIMEOUT_MS_CONFIG);
+      try {
+        // Read off as many messages as we can without triggering a timeout exception. The
+        // consumer timeout should be set very small, so the expectation is that even in the
+        // worst case, num_messages * consumer_timeout << request_timeout, so it's safe to only
+        // check the elapsed time once this loop finishes.
+
+        //FIXME :new api - just get all records for now..
+        while (iter.hasNext()){
+          //MessageAndMetadata<KafkaK, KafkaV> msg = iter.peek();
+          ConsumerRecordAndSize<ClientK, ClientV> recordAndSize = parent.createConsumerRecord(iter.next());
+          roughMsgSize = recordAndSize.getSize();
+          if (bytesConsumed + roughMsgSize >= maxResponseBytes) {
+            break;
+          }
+
+          //iter.next();
+          messages.add(recordAndSize.getRecord());
+          bytesConsumed += roughMsgSize;
+          // Updating the consumed offsets isn't done until we're actually going to return the
+          // data since we may encounter an error during a subsequent read, in which case we'll
+          // have to defer returning the data so we can return an HTTP error instead
+        }
+      } catch (ConsumerTimeoutException cte) {
```

Comment:
This exception shouldn't happen since the new consumer doesn't use an exception to indicate timeouts. Instead we should be checking for an empty list returned from poll().